### ISSUE TITLE
Outputs of the Hunyuan Video 1.5 in xDiT is different from Diffusers

### ISSUE THREAD

**Issue by cszhouY** [xdit-project/xDiT#655]
### Description
I compared the t2v outputs of the Hunyuan Video 1.5 using both Diffusers and xDiT, but the results are not identical. To my understanding, xDiT's implementation directly uses Diffusers' codebase. So, with the same random seed, the generated videos should be exactly the same—yet they are not.

|width|height|frames|steps|seed|
|-|-|-|-|-|
|1280|720|81|50|42|

prompt: A cat holding a paper with words "Hello, world!"
+ Diffusers 

https://github.com/user-attachments/assets/462b4cbf-aa55-4ff3-9909-18edb726abc6 

+ xDiT

https://github.com/user-attachments/assets/930101f1-c9c7-4112-aa43-606b5cda6907

### Code

+ Diffusers
```python
""" hunyuan video 1.5 example """
import torch
import time

dtype = torch.bfloat16
device = "cuda"

from diffusers import HunyuanVideo15Pipeline
from diffusers.utils import export_to_video

model_path =  "HunyuanVideo-1.5-Diffusers-720p_t2v"

pipe = HunyuanVideo15Pipeline.from_pretrained(
    model_path, 
    torch_dtype=dtype)
pipe.to("cuda")
pipe.vae.enable_tiling()

pipe.transformer.set_attention_backend("flash_varlen_hub")

prompt = 'A cat holding a paper with words "Hello, world!"'
seed = 42
height = 720
width = 1280
num_frames = 81
steps = 50

generator = torch.Generator(device=device).manual_seed(seed)

start_time = time.time()

video = pipe(
    prompt=prompt,
    generator=generator,
    num_frames=num_frames,
    height=height,
    width=width,
    num_inference_steps=steps,
).frames[0]

end_time = time.time()
elapsed_time = end_time - start_time
print(f"Elapsed time: {elapsed_time:.2f} seconds")

export_to_video(video, "hunyuan_video_15_base.mp4", fps=24)
```
+ xDiT
```python
import time
import torch
from xfuser.config.diffusers import has_valid_diffusers_version, get_minimum_diffusers_version

if not has_valid_diffusers_version("hunyuanvideo_15"):
    minimum_diffusers_version = get_minimum_diffusers_version("hunyuanvideo_15")
    raise ImportError(f"Please install diffusers>={minimum_diffusers_version} to use HunyuanVideo 1.5 models.")

from xfuser.model_executor.models.transformers.transformer_hunyuan_video15 import xFuserHunyuanVideo15Transformer3DWrapper
from diffusers import HunyuanVideo15Pipeline, HunyuanVideo15ImageToVideoPipeline
from diffusers import DiffusionPipeline
from diffusers.utils import export_to_video, load_image

from xfuser import xFuserArgs
from xfuser.config import FlexibleArgumentParser
from xfuser.core.distributed import (
    get_world_group,
    get_runtime_state,
    initialize_runtime_state,
)

def run_pipe(pipe: DiffusionPipeline, input_config, image, task):
    kwargs = {
        "prompt": input_config.prompt,
        "num_inference_steps": input_config.num_inference_steps,
        "num_frames": input_config.num_frames,
        "generator": torch.Generator(device="cuda").manual_seed(input_config.seed),
    }
    if task == "i2v":
        kwargs["image"] = image
    else: # t2v task
        kwargs["height"] = input_config.height
        kwargs["width"] = input_config.width
    return pipe(**kwargs).frames[0]

def main():
    parser = FlexibleArgumentParser(description="xFuser Arguments")
    parser.add_argument(
        "--task",
        type=str,
        required=True,
        choices=["i2v", "t2v"],
        help="The task to run."
    )
    args = xFuserArgs.add_cli_args(parser).parse_args()
    engine_args = xFuserArgs.from_cli_args(args)
    engine_config, input_config = engine_args.create_config()
    engine_config.runtime_config.dtype = torch.bfloat16
    local_rank = get_world_group().local_rank
    is_last_process = get_world_group().rank == get_world_group().world_size - 1

    image = None
    if args.task == "i2v":
        if not input_config.img_file_path:
            raise ValueError("Image file path must be provided for image-to-video task.")
        image = load_image(args.img_file_path)

    transformer = xFuserHunyuanVideo15Transformer3DWrapper.from_pretrained(
        engine_config.model_config.model,
        torch_dtype=torch.bfloat16,
        subfolder="transformer",
    )
    pipe = HunyuanVideo15Pipeline if args.task == "t2v" else HunyuanVideo15ImageToVideoPipeline
    pipe = pipe.from_pretrained(
        pretrained_model_name_or_path=engine_config.model_config.model,
        transformer=transformer,
        torch_dtype=torch.bfloat16,
    )

    pipe = pipe.to(f"cuda:{local_rank}")

    if args.enable_tiling:
        pipe.vae.enable_tiling()

    if args.enable_slicing:
        pipe.vae.enable_slicing()

    parameter_peak_memory = torch.cuda.max_memory_allocated(device=f"cuda:{local_rank}")

    initialize_runtime_state(pipe, engine_config)

    if engine_config.runtime_config.use_torch_compile:
        torch._inductor.config.reorder_for_compute_comm_overlap = True
        pipe.transformer = torch.compile(pipe.transformer, mode="default")

        # one full pass to warmup the torch compiler
        output = run_pipe(pipe, input_config, image, args.task)

    torch.cuda.reset_peak_memory_stats()
    start_time = time.time()

    output = run_pipe(pipe, input_config, image, args.task)

    end_time = time.time()

    elapsed_time = end_time - start_time
    peak_memory = torch.cuda.max_memory_allocated(device=f"cuda:{local_rank}")

    parallel_info = (
        f"ulysses{engine_args.ulysses_degree}_ring{engine_args.ring_degree}"
    )
    if input_config.output_type == "pil":
        if is_last_process:
            video_name = f"hunyuan_video_15_{args.task}_result_{parallel_info}_tc_{engine_args.use_torch_compile}_{input_config.height}x{input_config.width}.mp4"
            export_to_video(output, video_name, fps=24)
            print(f"video saved to {video_name}")

    if is_last_process:
        print(
            f"epoch time: {elapsed_time:.2f} sec, parameter memory: {parameter_peak_memory/1e9:.2f} GB, memory: {peak_memory/1e9:.2f} GB"
        )
    get_runtime_state().destroy_distributed_env()


if __name__ == "__main__":
    main()
```
```shell
PROMPT='A cat holding a paper with words "Hello, world!"'
torchrun --nproc_per_node=4 \
    hunyuan_video15_example.py \
    --model "/klxlake/public/models/HunyuanVideo-1.5-Diffusers-720p_t2v" \
    --task "t2v" \
    --ulysses_degree 4 \
    --num_inference_steps 50 \
    --warmup_steps 0 \
    --prompt "$PROMPT" \
    --height 720 \
    --width 1280 \
    --num_frames 81 \
    --seed 42 \
    --enable_tiling \
    --enable_model_cpu_offload
```

**Comment by avjves** [comment]
Fixed and merged in #656 :)
